fix: push tag: do not update remote ref

hhourani27 opened the pull request:
fixes #1900

I added comments below on my changes. You feedback is welcome!

hhourani27 commented on src/commands/push.js:
```diff
@@ -283,7 +283,12 @@ export async function _push({
+    remote &&
+    result.ok &&
+    result.refs[fullRemoteRef].ok &&
+    !fullRef.startsWith('refs/tags')
```
I used the same condition used in line 222 to check if ref == tag

hhourani27 commented on src/commands/push.js:
```diff
@@ -283,7 +283,12 @@ export async function _push({
+  ) {
     // TODO: I think this should actually be using a refspec transform rather than assuming 'refs/remotes/{remote}'
     const ref = `refs/remotes/${remote}/${fullRemoteRef.replace(
       'refs/heads',
```
This line assumes that ref == branch. So we should make sure that ref != tag

jcubic commented:
Looks good, are there any tests that check if the push of a tag is updating refs?

hhourani27 commented:
> Looks good, are there any tests that check if the push of a tag is updating refs?

Thanks. Pushing a tag should not update any ref in the local git repo.
I added a test on line 214 & 238 of `test-push.js` to make sure that the push tag is not creating a ref (reproducing the bug).
I also added a test on line 83 to make sure that pushing a branch does update the remote ref.

I'm open if you think we need more tests.
